src/gapstate.h

```diff
@@ -195,19 +199,22 @@ extern GAPState MainGAPState;
 
 #endif
 
-void InitMainGAPState(void);
+typedef void * GAPModuleState;
+typedef void (*GAPModuleConstructor)();
+typedef void (*GAPModuleDestructor)();
+
+typedef struct {
+    Int Offset;
+    GAPModuleConstructor Constructor;
+    GAPModuleDestructor Destructor;
+} GAPStateHandler;
```

Comment:
Why `GAPStateHandler`? To me, a "handler" is a function, or anything you can execute. That said, I don't have a better name right now, so feel free to keep it, it is trivial to rename it in the future, if somebody comes up with a better one :-).